apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -491,15 +482,16 @@ where
                             let tx_hash_key =
                                 replay_protection::get_tx_hash_key(&hash);
                             self.wl_storage
-                                .storage
                                 .delete(&tx_hash_key)
                                 .expect(
                                     "Error while deleting tx hash key from storage",
                                 );
+                            // Apply only to remove its hash,
+                            // since all other changes have already been dropped
+                            self.wl_storage.commit_tx();
```

Comment:
Why do we need to do this?

Reply:
it is to commit the removal of the `tx_hash_key` above

Reply:
Yes, it's a bit tricky. But we have to `commit_tx()` to apply changes on the write log when the following commit phase.